titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/TaskStatus.java

```diff
@@ -153,6 +153,14 @@ public static boolean isEvicted(Task task) {
                 ).orElse(false);
     }
 
+    public static boolean hasPod(Task task) {
+        TaskStatus acceptedStatus = JobFunctions.findTaskStatus(task, TaskState.Accepted).orElse(null);
```

Comment:
If there are many Accepted entries in the status history then you might not get the correct one.